python/ray/util/metrics.py

```diff
@@ -0,0 +1,185 @@
+            self._tag_cache = {}
+
+        self._tag_cache = tags
+        return self
+
+    def record(self, value: float) -> None:
+        """Record the metric point of the metric.
+
+        Args:
+            value(float): The value to be recorded as a metric point.
+        """
+        assert self._metric is not None
+        default_tag_copy = self._default_tags.copy()
+        default_tag_copy.update(self._tag_cache)
+        self._metric.record(value, tags=default_tag_copy)
+        # We should reset tag cache. Otherwise, with_tags will be corrupted.
+        self._tag_cache = {}
+
+    @property
+    def info(self) -> Dict[str, Any]:
+        """Return the information of this metric.
+
+        Example:
+            >>> counter = Counter("name", description="desc")
+                print(counter.info)
+                \"""
+                {
+                    "name": "name",
+                    "description": "desc"
+                    "tag_keys": ("ray.key")
+                    "default_tags": {"ray.key": "abc"}
+                }
+                \"""
+        """
+        return {
+            "name": self._name,
+            "description": self._description,
+            "tag_keys": self._tag_keys,
+            "default_tags": self._default_tags
+        }
+
+
+class Count(Metric):
+    """The count of the number of metric points.
+
+    This is corresponding to Prometheus' Count metric.
+
+    Args:
+        name(str): Name of the metric.
+        description(str): Description of the metric.
+        tag_keys(tuple): Tag keys of the metric.
+        tags(dict): Dictionary of default tag values.
+    """
+
+    def __init__(self,
+                 name: str,
+                 description: str = "",
+                 tag_keys: Tuple[str] = None,
+                 default_tags: Dict[str, Optional[str]] = None):
+        super().__init__(name, description, tag_keys, default_tags)
+        self._metric = CythonCount(self._name, self._description, self._unit,
+                                   self._tag_keys)
+
+
+class Histogram(Metric):
+    """Histogram distribution of metric points.
+
+    This is corresponding to Prometheus' Histogram metric.
+    Recording metrics with histogram will enable you to import
+    min, mean, max, 25, 50, 95, 99 percentile latency.
+
+    Args:
+        name(str): Name of the metric.
+        description(str): Description of the metric.
+        boundaries(list): Boundaries of histogram buckets.
+        tag_keys(tuple): Tag keys of the metric.
+        tags(dict): Dictionary of default tag values.
+    """
+
+    def __init__(self,
+                 name: str,
+                 description: str = "",
+                 boundaries: List[float] = None,
+                 tag_keys: Tuple[str] = None,
+                 default_tags: Dict[str, Optional[str]] = None):
+        super().__init__(name, description, tag_keys, default_tags)
+        self.boundaries = boundaries if boundaries else []
```

Comment:
- is empty boundaries defined behavior?
- `boundaries or []`

Reply:
That's a good question. I will raise an exception when the boundaries are empty instead.